hyperspy/model.py

```diff
@@ -891,6 +899,312 @@ def _model_function(self, param):
         to_return = self.__call__(non_convolved=False, onlyactive=True)
         return to_return
 
+    @property
+    def nonlinear_parameters(self):
+        "List all active nonlinear parameters"
+        nonlinear = []
+        for comp in self:
+            if comp.active:
+                for para in comp.parameters:
+                    if not para._is_linear:
+                        nonlinear.append(para)
+        return nonlinear
+
+    @property
+    def linear_parameters(self):
+        "List all active linear parameters"
+        linear = []
+        for comp in self:
+            if comp.active:
+                for para in comp.parameters:
+                    if para._is_linear:
+                        linear.append(para)
+        return linear
+
+    def _set_linear_parameters_to_one(self, set_map=False):
```

Comment:
Agreed, it can be directly in the code. 
(I believe you understand the reason, but for clarity and as an example):
```python
# Linear parameters must be set to a nonzero value before fitting to avoid the entire 
# component being zero. We choose the value of 1 for no particular reason.
```
Here we *could* check for nonzero components and change only them, but I decided it was less code to do it this way.